cmake/EthCompilerSettings.cmake

```diff
@@ -41,15 +41,6 @@ if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" MA
     set(CMAKE_CXX_FLAGS_RELEASE        "-O3 -DNDEBUG")
     set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g")
 
-    option(USE_LD_GOLD "Use GNU gold linker" ON)
```

Comment:
Out of curiosity, which linker will we use for clang builds if we don't use the gold linker?

Reply:
The GNU `ld`. Also `lld` from LLVM can be used (is much faster, I'm using it for building LLVM itself where linking time is huge). In the same way: `-fuse-ld=lld`. But only clang supports it (GCC not).

As I explained in the commit message, Google has switched to use LLD, and they are not interested in maintaining ld.gold. Fedora is considering moving ld.gold out of the main package repo. And I have linking error when using clang-8. There must be new "linking features" that clang-8 depends on.

Reply:
Thanks for the clarification!